app/src/main/java/org/piwigo/ui/settings/SettingsActivity.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Piwigo for Android
+ * Copyright (C) 2019-2019 Piwigo Team http://piwigo.org
+ * Copyright (C) 2019-2019 Radko Varchola
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package org.piwigo.ui.settings;
+
+import android.os.Bundle;
+import android.preference.ListPreference;
+import android.preference.PreferenceActivity;
+import android.view.LayoutInflater;
+import android.widget.LinearLayout;
+
+import org.piwigo.R;
+import org.piwigo.io.repository.PreferencesRepository;
+
+import javax.inject.Inject;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.widget.Toolbar;
+import dagger.android.AndroidInjection;
+
+
+public class SettingsActivity extends PreferenceActivity {
+
+    @Inject
+    PreferencesRepository preferences;
+
+    ListPreference mPreferencePhotosPerRow;
+    ListPreference mPreferenceThumbnailSize;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        AndroidInjection.inject(this);
+        super.onCreate(savedInstanceState);
+        addPreferencesFromResource(R.xml.settings_preferences);
```

Comment:
this is deprecated. Sorry in case you took it from the site I referred you to, but for sure, if something is out-of-date we should not just take is because if worked, but check which alternative is there instead...
In this case we should use an PreferenceFragementCompat as explained in the [official docs](https://developer.android.com/guide/topics/ui/settings.html#inflate_the_hierarchy)

Reply:
Ok, will change. I was thinking that exactly this solution you want.

Reply:
Sorry if I left a wrong impression here. It was just a guide that I found to give you something.g at hand showing the preferences in xml and the summary etc.